### PR TITLE
BC-6188: fix components/templates tests

### PR DESCRIPTION
**Pull request by uidp**
# Short Description
Vue3 upgrade: fix components/templates tests

## Links to Ticket and related Pull-Requests
BC-6188

## Checklist before merging

- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [ ] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/templates/DefaultWireframe.unit.js [hunk 1]
```diff
@@ -1,36 +1,44 @@
...
 
 describe("DefaultWireframe", () => {
 	it("shows title", () => {
 		const wrapper = mount(DefaultWireframe, {
-			...createComponentMocks({}),
-			propsData: { fullWidth: true, headline: "dummy titel" },
+			global: {
+				plugins: [createTestingVuetify(), createTestingI18n()],
+			},
+			props: { fullWidth: true, headline: "dummy titel" },
```
Comment (MartinSchuhmacher) [inline src/components/templates/DefaultWireframe.unit.js:14]: just a small thing, but for consistency I would prefer `dummy title`

File: src/components/templates/DefaultWireframe.unit.js [hunk 2]
```diff
@@ -1,36 +1,44 @@
...
 		});
 
 		const h1 = wrapper.find("h1");
 		expect(h1.text()).toBe("dummy titel");
 	});
 	it("shows breadcrumbs", () => {
 		const wrapper = mount(DefaultWireframe, {
-			...createComponentMocks({}),
-			propsData: {
+			global: {
```
Comment (odalys-dataport) [inline src/components/templates/DefaultWireframe.unit.js:22]: Maybe we could move mounting into a separate function

File: src/components/templates/RoomDashboard.unit.ts [hunk 3]
```diff
@@ -301,7 +312,7 @@ describe("@/components/templates/RoomDashboard.vue", () => {
 			expect(wrapper.vm.isDragging).toBe(false);
 			cardElement.vm.$emit("on-drag");
 			expect(wrapper.vm.isDragging).toBe(true);
-			await wrapper.vm.$nextTick();
+			await nextTick();
```
Comment (odalys-dataport) [inline src/components/templates/RoomDashboard.unit.ts:315]: Did you check if these are necessary? (Meaning every occurance of nextTick)

File: src/components/templates/RoomWrapper.unit.ts [hunk 4]
```diff
@@ -72,63 +78,52 @@ const mockData = [
...
+				const wrapper = getWrapper({
 					props: { hasRooms: false },
-					computed: { $mq: () => "desktop", isLoading: () => false },
 				});
 
 				expect(
 					wrapper.findComponent({ ref: "rooms-empty-state" }).exists()
 				).toBe(true);
-
-				wrapper.destroy();
```
Comment (odalys-dataport) [inline src/components/templates/RoomWrapper.unit.ts:115]: Just out of curiosity, why is this not necessary anymore?

File: src/components/templates/RoomWrapper.unit.ts [hunk 5]
```diff
@@ -183,59 +178,58 @@ describe("@templates/RoomWrapper.vue", () => {
...
-			const wrapper = getWrapper();
-
-			const fabComponent = wrapper.find(".wireframe-fab");
-			expect(fabComponent.exists()).toBe(false);
-
-			wrapper.destroy();
-		});
-	});
+	// VUE3_UPGRADE disabled until we have a v-custom-fab replacement
+	// describe("when user does not have course create permission", () => {
```
Comment (odalys-dataport) [inline src/components/templates/RoomWrapper.unit.ts:201]: Could we mark this in the ticket for v-speed-dial replacement, so this tests gets fixed within that ticket, please?

File: src/components/templates/TasksDashboardMain.unit.ts [hunk 6]
```diff
@@ -1,4 +1,3 @@
-/* eslint-disable max-lines */
```
Comment (odalys-dataport) [inline src/components/templates/TasksDashboardMain.unit.ts:1]: Oh do we not have this rule anymore?

**Review by odalys-dataport — CHANGES_REQUESTED** [review]

File: src/components/templates/DefaultWireframe.unit.js (hunk 1 above)
Comment (uidp) [reply src/components/templates/DefaultWireframe.unit.js:14]: 😱 that's ugly. fixed.

File: src/components/templates/DefaultWireframe.unit.js (hunk 2 above)
Comment (uidp) [reply src/components/templates/DefaultWireframe.unit.js:22]: Hm, the different setups are quiet different, I didn't want to do much refactoring. That's why I left the code as is (for now).

File: src/components/templates/RoomDashboard.unit.ts (hunk 3 above)
Comment (uidp) [reply src/components/templates/RoomDashboard.unit.ts:315]: Yes, it's necessary.

File: src/components/templates/RoomWrapper.unit.ts (hunk 4 above)
Comment (uidp) [reply src/components/templates/RoomWrapper.unit.ts:115]: The wrapper is destroyed automatically when it gets out of scope (almost like in the runtime code).

File: src/components/templates/RoomWrapper.unit.ts (hunk 5 above)
Comment (uidp) [reply src/components/templates/RoomWrapper.unit.ts:201]: Added information in https://ticketsystem.dbildungscloud.de/browse/BC-6254

**Review by MartinSchuhmacher — APPROVED** [review]

File: src/components/templates/TasksDashboardMain.unit.ts (hunk 6 above)
Comment (odalys-dataport) [reply src/components/templates/TasksDashboardMain.unit.ts:1]: Yea I think so

**Review by odalys-dataport — APPROVED** [review]

**Pull request merged**